core/http-auth-aws/src/main/java/software/amazon/awssdk/http/auth/aws/internal/signer/RollingSigner.java

```diff
@@ -0,0 +1,54 @@
+/*
+ * Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License").
+ * You may not use this file except in compliance with the License.
+ * A copy of the License is located at
+ *
+ *  http://aws.amazon.com/apache2.0
+ *
+ * or in the "license" file accompanying this file. This file is distributed
+ * on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+ * express or implied. See the License for the specific language governing
+ * permissions and limitations under the License.
+ */
+
+package software.amazon.awssdk.http.auth.aws.internal.signer;
+
+import static software.amazon.awssdk.http.auth.aws.util.SignerUtils.computeSignature;
+import static software.amazon.awssdk.utils.BinaryUtils.toHex;
+
+import java.util.function.Function;
+import software.amazon.awssdk.annotations.SdkInternalApi;
+
+/**
+ * A class which calculates a rolling signature of arbitrary data using HMAC-SHA256.
+ * Each time a signature is calculated, the prior calculation is incorporated, hence "rolling".
+ */
+@SdkInternalApi
+public final class RollingSigner {
```

Comment:
What's this used for?

Reply:
For calculating signatures of chunks. Each chunk signature depends on the last.

Reply:
I guess my question was more cosmic. It's only used in tests. What is it used for outside of tests?

Reply:
It will be used in the S3 signer

Reply:
Okay, I'll ignore it for now, then.

Reply:
The test is an example of how AWS chunked encoding would be configured.